### PR TITLE
[Agency Dashboard] Change ALLOW_LIST to DENY_LIST

### LINKED ISSUE #898: [Dashboard] Change agency allowlist to a denylist
Moving forward, all agencies that are onboarded onto Publisher should have access to the dashboard. Thus, it makes more sense to have a denylist for dashboard access rather than the allowlist that we currently have. The new denylist should include all current agencies with publisher accounts minus the agencies currently on the allowlist.

### PR DESCRIPTION
**Pull request by mxosman**
## Description of the change

The decision point made after a discussion with CSG revisiting the purpose of the allow-list was to convert the current allow-list and logic to be a deny-list. All new agencies will by default have a live, visible dashboard. The existing agencies are the ones we are concerned about since they have not explicitly opt'd in to having their dashboards be live and visible to the public. Those agencies will be on the deny-list which will prevent their dashboards from being visible until they explicitly decide to go live. 

Note: this is a temporary solution (until a backend solution is implemented) that will help us handle requests to allow-list an agency by there being less things to update (unless a currently deny-listed agency wants to go live) and avoiding the tight coupling of a frequently updated allow-list and our production deploys (since new agencies should be allow-listed by default now). Hopefully this buys us some time and requires less updates to the hardcoded list until we scope and implement a backend solution.

https://github.com/Recidiviz/justice-counts/assets/59492998/b5be316b-5423-414f-8434-d9113466999d

## Related issues

Closes #898

### Development

**This box MUST be checked by the submitter prior to merging**:
- [x] **Double- and triple-checked that there is no Personally Identifiable Information (PII) being mistakenly added in this pull request**

These boxes should be checked by the submitter prior to merging:
- [ ] Tests have been written to cover the code changed/added as part of this pull request

### Code review

These boxes should be checked by reviewers prior to merging:

- [ ] This pull request has a descriptive title and information useful to a reviewer
- [ ] This pull request has been moved out of a Draft state, has no "Work In Progress" label, and has assigned reviewers
- [ ] Potential security implications or infrastructural changes have been considered, if relevant

### CODE REVIEW

File: agency-dashboard/src/utils/allowlist.ts [hunk 1]
```diff
@@ -17,16 +17,591 @@
...
+  "adams county sheriff's office": true,
+  "administrative office of the illinois courts": true,
+  "alabama department of corrections": true,
+  "alaska department of corrections": true,
+  "albany village police dept": true,
+  "allen county sheriff's office": true,
+  "amanda police department": true,
+  "amelia police department": true,
+  "american township police department": true,
+  "ames police department": false,
```
Comment (mxosman) [inline agency-dashboard/src/utils/allowlist.ts:47]: This one was just requested to be allowlisted by CSG today.

File: agency-dashboard/src/utils/allowlist.ts [hunk 2]
```diff
@@ -17,16 +17,568 @@
...
+  "worthington police department",
+  "wyoming department of corrections",
+  "yavapai county district attorney",
+];
 
-export const isAllowListed = ({ name }: UserAgency): boolean | undefined =>
-  ALLOW_LIST[name];
+export const isDenyListed = ({ name }: UserAgency): boolean | undefined => {
+  const lowercaseName = name.toLocaleLowerCase();
+  return DENY_LIST.includes(lowercaseName);
```
[the remark this answers is not in the document]
Comment (morden35) [reply agency-dashboard/src/utils/allowlist.ts:583]: I think you can get around this by making this a set rather than a list

**Review by morden35 — APPROVED** [review]

File: agency-dashboard/src/utils/allowlist.ts [hunk 3]
```diff
@@ -17,16 +17,568 @@
...
+  "west virginia division of corrections and rehabilitation",
+  "whitehall division of police",
+  "wilmington police department",
+  "winchester police department",
+  "wisconsin department of justice",
+  "wittenberg university police division",
+  "woodmere village police department",
+  "worthington police department",
+  "wyoming department of corrections",
+  "yavapai county district attorney",
```
Comment (morden35) [inline agency-dashboard/src/utils/allowlist.ts:578]: out of curiosity, how did you generate this list?

File: agency-dashboard/src/utils/allowlist.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (mxosman) [reply agency-dashboard/src/utils/allowlist.ts:583]: Ah - of course! 🤦  That's such a great idea, Michelle! Thank you so much. Where’s my brain today? Let me do that real quick!

[the remark this answers is not in the document]
Comment (morden35) [reply agency-dashboard/src/utils/allowlist.ts:583]: its all good!

File: agency-dashboard/src/utils/allowlist.ts (hunk 3 above)
Comment (mxosman) [reply agency-dashboard/src/utils/allowlist.ts:578]: Oh - I probably should've double checked this with you! So, I logged into the prod DB and ran `select name from source;`

This produced a long list that was really hard to copy - so I found this helpful SQL command that will turn it all into a comma-separated string that I could take and transform into these strings. [I'm sure there was much easier/straightforward/better way to do this]

```
SELECT string_agg(name, ', ') AS concatenated_values FROM source;
```

Comment (morden35) [reply agency-dashboard/src/utils/allowlist.ts:578]: yeah that should work!

File: agency-dashboard/src/utils/denylist.ts [hunk 4]
```diff
@@ -0,0 +1,584 @@
...
+  "wittenberg university police division",
+  "woodmere village police department",
+  "worthington police department",
+  "wyoming department of corrections",
+  "yavapai county district attorney",
+]);
+
+export const isDenyListed = ({ name }: UserAgency): boolean | undefined => {
+  const lowercaseName = name.toLocaleLowerCase();
+  return DENY_LIST.has(lowercaseName);
```
Comment (mxosman) [inline agency-dashboard/src/utils/denylist.ts:583]: Confirming this works exactly the same while testing

File: agency-dashboard/src/utils/allowlist.ts (hunk 3 above)
Comment (mxosman) [reply agency-dashboard/src/utils/allowlist.ts]: Phew - thank you for asking! I realized I should've probably double checked that I was getting the right list.

**Comment by mxosman** [comment]
Thank you so much, Michelle!

**Pull request merged**